### ISSUE TITLE
[Other] Fix browser inclusion in binary for all platforms

### ISSUE THREAD

**Issue by justinkambic** [elastic/synthetics-recorder#150]
### Task summary

Today, when we run our release script, it works quite well. Electron builder outputs binaries for Windows, Linux, and macOS.

Unfortunately, we have some custom scripting in our build process responsible for packaging the correct browser version as part of the process. This only works for the current platform though, so if you build all three binaries on a Mac, only the Mac version of the browser gets baked into the DMG. The Windows and Linux binaries will fail to run chromium in a recording session, because the browser is missing.

### Linked resources

The script responsible for installing Playwright: https://github.com/elastic/synthetics-recorder/blob/main/scripts/install-pw.js.

### Additional information

It's possible to work around this problem by building binaries for each platform natively. The drawback is it fragments the process and requires access to multiple build machines/CI processes.

**Comment by vigneshshanmugam** [comment]
We could utilize Jenkins workers on all 3 platforms to build and upload them to GH. I was talking about this today with @kyungeunni to avoid doing it manually for now.

**Comment by justinkambic** [comment]
Yes this would be a valid workaround. Transfers the complexity from electron-builder configuration to CI but we probably know the latter world a bit better. It might make it easier to integrate code signing on a per-platform basis as well.

**Comment by kyungeunni** [comment]
I was pondering about this and not sure if it's the right way to go setting up Jenkins workers for all platforms and build. (An example: we need to also release a build for Apple Silicon, would we need to set up a Jenkins worker for that just because we need to download the right browser?)

Possible alternatives:
- download all the browsers for all platforms, and configure the electron builder to include the applicable browser for each build using [per platform option] - `extraResources`
- do not include the browser in the builds, if the user running it the recorder for the first time, let it download the right browser

**Comment by kyungeunni** [comment]
As a proof of concept, I've manually downloaded a browser for windows on my Mac, placed it in `local-browsers/chromium-${REVISION}/chrome-win` directory(this is the path Playwright is going to find the browser executable), then configured the electron-builder to include that directory when building the binary for windows. I've tested it in the virtual machine and seem to be working:
<img width="512" alt="Screenshot 2022-02-22 at 16 41 26" src="https://user-images.githubusercontent.com/16660866/155167912-146e6dc5-a3df-40e7-82ed-4b89e833da58.png">
 
Here's the electron-builder configuration I used for the PoC: https://github.com/kyungeunni/synthetics-recorder/pull/2

**Comment by justinkambic** [comment]
> we need to also release a build for Apple Silicon, would we need to set up a Jenkins worker for that just because we need to download the right browser?

I tend to agree with this sentiment; as a long-term solution we will want to be able to issue one command and have the release built for all supported platforms via `electron-builder`. This browser constraint is the only thing stopping us doing that.

I think a good temporary fix is to create a `prebuild` script like @kyungeunni suggests, and then eventually see if we can commit an enhancement upstream in Playwright to allow us to request specifically which browsers we want to download. At that point we could delete our custom script and rely on the PW feature.

> I've tested it in the virtual machine and seem to be working

va bene 😄

**Comment by kyungeunni** [comment]
We were able to tackle down building binaries for mac and windows but encountered an issue with Linux. @justinkambic
 tried to run the binary and run it in dev mode but none of them worked. It all failed when running the playback(pressing Test button).
 The error we observed is not due to native dependencies or browser not being available so it's not strictly related to this issue, but  I'll leave this comment here as it's still build-related issue.
